File: 14-longest-common-prefix/14-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    for (let i = 0; i < strs[0].length; i++) {
        let char = strs[0].charAt(i);
        for (let j = 0; j < strs.length; j++) {
            if (i === strs[j].length || strs[j].charAt(i) !== char) {
                return strs[0].slice(0, i);
            }
        }
    }
    return strs[0];
};

File: 14-longest-common-prefix/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0: return ""
        for i in range(len(strs[0])):
            # char is set to iterative letter of first string
            char = strs[0][i]
            for j in range(len(strs)):
                # if iterator equals length of current string or if current letter does not equal char 
                # return prefix up to this point
                if i == len(strs[j]) or strs[j][i] != char:
                    return strs[0][:i]
        return strs[0]
        